### PR TITLE
RED-377 Consolidate RPC clients to monorepo

### PR DESCRIPTION
**Pull request by danielemiliogarcia**
## Description
We’re using the same RPC client code in multiple repositories. Let’s move this code into the monorepo so it can be exported for use in other packages.

The RPC clients can be moved into the redshift-utils package. Ensure that we move all RPC methods that are present in all three repos.

## Related PRs
https://github.com/RadarTech/redshift-backend/pull/111
https://github.com/RadarTech/starboard-js/pull/171

## Checks
- [x] Tests were run locally
- [x] Please identify two developers to review this change
- [x] Linting was done locally
- [x] Code and unit of code written has an associated test
- [x] If readme needs to be updated to reflect this unit of work

## Impacted Areas in Application
Starboard, Redshift, monorepo

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -15,16 +15,20 @@
...
     "lerna": "^2.9.0",
     "lint-staged": "^8.1.0",
     "mocha": "^5.2.0",
     "prettier": "^1.15.3",
     "rimraf": "^2.6.2",
+    "satoshi-bitcoin-ts": "^0.2.4",
     "tslint": "^5.11.0",
-    "typescript": "^3.2.1"
+    "typescript": "^3.2.1",
+    "value-utils": "^0.1.0"
```
Comment (cavanmflynn) [inline package.json]: These should not be added in the top-level package.json and they should be `dependencies`

File: packages/htlc/package.json [hunk 2]
```diff
@@ -13,6 +13,7 @@
   "license": "MIT",
   "dependencies": {
     "@radar/redshift-types": "^0.5.11",
+    "@radar/redshift-utils": "^0.1.6",
```
Comment (cavanmflynn) [inline packages/htlc/package.json]: Should be a dev dependency if it's only used in tests

File: packages/redshift-utils/src/rpc/base-rpc-client.ts [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+  abstract async getFeeEstimate(): Promise<BigNumber>;
+
+  /**
+   * Instantiate a new rpc client
+   * @param network The rpc client network
+   * @param subnet The rpc client subnet
+   */
+  constructor(
+    network: Network,
+    subnet: Subnet,
```
Comment (cavanmflynn) [inline packages/redshift-utils/src/rpc/base-rpc-client.ts]: Why keep the network and subnet params if we're forcing the user to pass the connection config?

File: packages/redshift-utils/src/rpc/ethereum/ethereum-rpc-client.ts [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+
+export class EthereumRpcClient extends BaseRpcClient {
+  /**
+   * Instantiate a new ethereum rpc client
+   * @param network The rpc client network
+   * @param subnet The rpc client subnet
+   */
+  constructor(
+    network: Network,
+    subnet: Subnet,
```
Comment (cavanmflynn) [inline packages/redshift-utils/src/rpc/ethereum/ethereum-rpc-client.ts]: Same question. Why keep these?

File: packages/redshift-utils/src/rpc/base-rpc-client.ts (hunk 3 above)
Comment (danielemiliogarcia) [reply packages/redshift-utils/src/rpc/base-rpc-client.ts]: yeap, that make sense, as it was assigning the value to the protected class var the IDE didn't mark that grayed out, but its not used anymore

File: packages/redshift-types/src/rpc.ts [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+   * @param Hex string representation of the transaction
+   */
+  sendRawTransaction(txHex: string): Promise<any>;
+
+  /**
+   * Fetch a fee estimate for submitting transactions to a respective network
+   * NOTE: the data returned is contextual to the network
+   */
+  getFeeEstimate(): Promise<BigNumber>;
+}
```
Comment (cavanmflynn) [inline packages/redshift-types/src/rpc.ts]: This is directly coupled to the RPC client. I don't think it belongs in this package.

Comment (cavanmflynn) [reply packages/redshift-types/src/rpc.ts]: Thinking about this more, these are all abstract methods on the base rpc client class so this interface is unnecessary and can be removed completely.

Comment (danielemiliogarcia) [reply packages/redshift-types/src/rpc.ts]: ill move it, i thought we want all interfaces in types package

File: packages/redshift-types/package.json [hunk 6]
```diff
@@ -11,7 +11,8 @@
   "dependencies": {
     "@types/bip32": "^1.0.1",
     "@types/bitcoinjs-lib": "4.0.0",
-    "@types/web3": "^1.0.18"
+    "@types/web3": "^1.0.18",
+    "bignumber.js": "^8.0.2"
```
Comment (cavanmflynn) [inline packages/redshift-types/package.json]: Remove `bignumber.js`

**Comment by cavanmflynn** [comment]
I missed this before, but `rpc-config.ts` should not be included in the redshift-utils. That should only be used in the tests.

**Comment by danielemiliogarcia** [comment]
> rpc-config.ts

on that

**Pull request merged**